docs/syntax_and_semantics/generics.md

```diff
@@ -91,3 +91,69 @@ end
 class GenericChild(T) < Parent(T)
 end
 ```
+
+## Generics with variable number of arguments
+
+We may define a Generic class with a variable number of arguments using the [splat operator](./operators.md#splats).
+
+Let's see an example where we define a Generic class called `Foo` and then we will use it with two type parameters `Int32` and `String`:
+
+```crystal-play
+class Foo(*T)
+  getter content
+
+  def initialize(*@content : *T)
+  end
+end
+
+foo = Foo(Int32, String).new(42, "Life, the Universe, and Everything")
```

Comment:
Can we also add examples with inferrred argument types, and a different arity of generic arguments? (could be in a single example)